File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/InputMappingIntSetting.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.model;

import androidx.annotation.NonNull;

import org.dolphinemu.dolphinemu.features.input.model.controlleremu.NumericSetting;
import org.dolphinemu.dolphinemu.features.settings.model.AbstractIntSetting;
import org.dolphinemu.dolphinemu.features.settings.model.Settings;

public class InputMappingIntSetting implements AbstractIntSetting
{
  private final NumericSetting mNumericSetting;

  public InputMappingIntSetting(NumericSetting numericSetting)
  {
    mNumericSetting = numericSetting;
  }

  @Override
  public int getInt()
  {
    return mNumericSetting.getIntValue();
  }

  @Override
  public void setInt(@NonNull Settings settings, int newValue)
  {
    mNumericSetting.setIntValue(newValue);
  }

  @Override
  public boolean isOverridden()
  {
    return false;
  }

  @Override
  public boolean isRuntimeEditable()
  {
    return true;
  }

  @Override
  public boolean delete(@NonNull Settings settings)
  {
    mNumericSetting.setIntValue(mNumericSetting.getIntDefaultValue());
    return true;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/InputMappingIntSetting.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.model

import org.dolphinemu.dolphinemu.features.input.model.controlleremu.NumericSetting
import org.dolphinemu.dolphinemu.features.settings.model.AbstractIntSetting
import org.dolphinemu.dolphinemu.features.settings.model.Settings

class InputMappingIntSetting(private val numericSetting: NumericSetting) : AbstractIntSetting {
    override val int: Int
        get() = numericSetting.getIntValue()

    override fun setInt(settings: Settings, newValue: Int) = numericSetting.setIntValue(newValue)

    override val isOverridden: Boolean = false

    override val isRuntimeEditable: Boolean = true

    override fun delete(settings: Settings): Boolean {
        numericSetting.setIntValue(numericSetting.getIntDefaultValue())
        return true
    }
}
